draft-irtf-cfrg-vdaf.md

```diff
@@ -3663,37 +3663,56 @@ way. The parameters for this circuit are summarized below.
 
 ### Prio3Sum
 
-| Parameter         | Value                                      |
-|:------------------|:-------------------------------------------|
-| `Valid`           | `Sum(Field128, bits)` (this section)       |
-| `Field`           | `Field128` ({{fields}})                    |
-| `PROOFS`          | `1`                                        |
-| `Xof`             | `XofTurboShake128` ({{xof-turboshake128}}) |
+| Parameter         | Value                                          |
+|:------------------|:-----------------------------------------------|
+| `Valid`           | `Sum(Field64, max_measurement)` (this section) |
+| `Field`           | `Field64` ({{fields}})                         |
+| `PROOFS`          | `1`                                            |
+| `Xof`             | `XofTurboShake128` ({{xof-turboshake128}})     |
 {: title="Parameters for Prio3Sum."}
 
 The next instance of Prio3 supports summing of integers in a pre-determined
-range. Each measurement is an integer in range `[0, 2^bits)`, where `bits` is an
-associated parameter.
+range. Each measurement is an integer in `range(max_measurement+1)`, where
+`max_measurement` is an associated parameter equal to the largest valid
+measurement.
+
+The range check is accomplished by encoding the measurement in a way that
+ensures it is in range, then comparing the reported measurement to the range
+checked measurement. Let
+
+* `bits = max_measurement.bit_length()`, the number of bits needed to encode
+  the largest valid measurement
+
+* `offset = 2**bits - 1 - max_measurement`
+
+The range checked measurement is the bit-encoding of `offset` plus the
+measurement. Observe that only measurements between `0` and `max_measurement`
+inclusive can be encoded this way with `bits` bits.
 
-The validity circuit is denoted `Sum`. The measurement is encoded as a
-length-`bits` vector of field elements, where the `l`th element of the vector
-represents the `l`th bit of the summand:
+The circuit, denoted `Sum`, first checks that each entry of this bit vector is a
+one or a zero. It then decodes it and subtracts it from `offset` plus the
+reported value. Since the range checked measurement is in the correct range,
+equality implies that the reported measurement is as well.
```

Comment:
Continued
```suggestion
The circuit, denoted `Sum`, first checks that each entry of both bit vectors is a
one or a zero. It then decodes both the measurement and the offset
measurement, and subtracts `offset` from the latter. It then checks if these two
values are equal. Since both the measurement and the measurement plus `offset`
are in the same range of `0` to `2**bits - 1`, this means that the measurement
itself is between `0` and `max_measurement`.
```